File: basics/ZeroToY.java
import java.util.Scanner;
public class ZeroToY {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-->0)
        {
            int x = sc.nextInt();
            int y =sc.nextInt();
            int steps = 0;
            int[] pow = new int[10];
            pow[0] = 1;
            for(int i=1;i<10;i++)
            pow[i] = pow[i-1] * 10; //2,7
            while(y!=0)
            {
              steps++;
              if(y>x)
              {
                System.out.println(y);
                for(int i=9;i>=0;i--)
                {
                    System.out.println(y+" "+(x*pow[i]));
                    if(y>(x*pow[i]))
                    {
                        y -= (x*pow[i]);
                        break;
                    }
                }
                System.out.println(y);
              }
              else{
                y--;
              }
            }
            System.out.println(steps);
        }
        sc.close();
    }
}

File: basics/ZeroToY.py
def main():
    t = int(input())
    for _ in range(t):
        xy = list(map(int,input().split()))
        x = xy[0]# 3 42
        y = xy[1]
        steps = 0
        pow_ = [1] * 10
        for i in range(1, 10):
            pow_[i] = pow_[i - 1] * 10

        while y != 0:
            
            if y >= x:
                for i in range(9, -1, -1):
                    if y >= (x * pow_[i]):
                        y -= (x * pow_[i])
                        break
            else:
                steps+=y
                break
            
            steps += 1
        print(steps)

if __name__ == "__main__":
    main()
